### PR TITLE
Area filtering

### LINKED ISSUE #170: error checking with proteogenomics data
better edge case errors for proteogenomic data

1. sample with only 1 analyte in data `!all(names(data) %in% names(probeAssay))`

### LINKED ISSUE #171: bug in setSegmentQCFlags() minArea filtering
When below code is run on protein data, segments with area greater than 20 are flagged as failing QC because of low area. 
GeomxTools v3.0.1

```
 proteinData <- setSegmentQCFlags(proteinData, qcCutoffs = list(percentSaturation = 45,
                                                               minSegmentReads=1000, 
                                                               percentAligned=80, 
                                                               minNegativeCount=10, 
                                                               maxNTCCount=60, 
                                                               minNuclei=16000, 
                                                               minArea=20))
```

[SPG_protein_data.zip](https://github.com/Nanostring-Biostats/GeomxTools/files/8689926/SPG_protein_data.zip)

### PR DESCRIPTION
**Pull request by maddygriz**
Differences in files are hard to see with the change in whitespace characters.

updates are on lines:
34 (ellipse)
115-122 (added lines), resolves #171
195-206 (added lines), resolves #170

### CODE REVIEW

**Comment by NicoleEO** [comment]
Can you change this to merge into dev instead?

File: R/readNanoStringGeoMxSet.R [hunk 1]
```diff
@@ -1,331 +1,351 @@
...
+              paste0(missingPhenoData, sep=", "),
+              "These will be excluded from the GeoMxSet object.")
+    }
+    pheno[[j]] <- NULL
+    if (phenoDataColPrefix != "") {
+      colnames(pheno) <- paste0(phenoDataColPrefix, colnames(pheno))
+      protocolDataColNames <- paste0(phenoDataColPrefix, protocolDataColNames)
+    }
+    
+    if("area" %in% colnames(pheno)){
```
Comment (fnew) [inline R/readNanoStringGeoMxSet.R:116]: Could we possibly generalize this for 'area' or 'Area', as well?

File: R/readNanoStringGeoMxSet.R [hunk 2]
```diff
@@ -1,331 +1,351 @@
...
+        pheno[["area"]] <- as.numeric(pheno[["area"]])
+    }
+    if("nuclei" %in% colnames(pheno)){
+        pheno[["nuclei"]] <- as.numeric(pheno[["nuclei"]])
```
Comment (fnew) [inline R/readNanoStringGeoMxSet.R:120]: I've also seen columns like "AOINucleiCount" or "Nuclei"

File: R/readNanoStringGeoMxSet.R (hunk 1 above)
Comment (maddygriz) [reply R/readNanoStringGeoMxSet.R]: This is now changed case independent

File: R/readNanoStringGeoMxSet.R (hunk 2 above)
Comment (maddygriz) [reply R/readNanoStringGeoMxSet.R]: case-independent search of both "Nuclei" and "AOINucleiCount"

**Comment by maddygriz** [comment]
@fnew Have you been able to look at this after my changes?

**Comment by maddygriz** [comment]
@fnew Can you finish looking at this after my fixes?

File: R/readNanoStringGeoMxSet.R [hunk 3]
```diff
@@ -1,331 +1,357 @@
...
+    
+    if("area" %in% tolower(colnames(pheno))){
+      areaCol <- colnames(pheno)[which(tolower(colnames(pheno)) == "area")]
+      pheno[[areaCol]] <- as.numeric(pheno[[areaCol]])
+    }
+    if("nuclei" %in% tolower(colnames(pheno))){
+      nucleiCol <- colnames(pheno)[which(tolower(colnames(pheno)) == "nuclei")]
+      pheno[[nucleiCol]] <- as.numeric(pheno[[nucleiCol]])
+    }
+    if("AOINucleiCount" %in% tolower(colnames(pheno))){
```
Comment (fnew) [inline R/readNanoStringGeoMxSet.R:124]: I think this should always return FALSE since the colnames are set to lower case

**Review by fnew — APPROVED** [review]
Looks good!

**Pull request merged**